Add --dry-run flag for startup validation (#3772)

Grinsven opened the pull request:
Fixes #3772 - Added `--dry-run` flag to orderbook, autopilot, and driver to validate startup (config, DB, RPC, contracts) before deployment. Exits after initialization, before starting main loop.

Would have caught the `deploy()` vs `deployed()` typo from PR #3759 since exit point is after contract loading. Added CI smoke tests job that needs `FORK_URL` secret for full validation.

Grinsven commented:
I have read the CLA Document and I hereby sign the CLA

jmg-duarte commented on .github/workflows/pull-request.yaml:
```diff
@@ -229,6 +229,96 @@ jobs:
+          else
+            echo "Correctly failed with invalid database"
+            exit 0
+          fi
```
I don't think these are necessary

jmg-duarte reviewed (commented):
I'm not a fan of the dry-run name because it's not a true dry-run, it doesn't keep running and it's purpose is to ensure the initialization sequence goes well.

Besides that, everything seems ok. Keen to hear my teammates thoughts

Grinsven commented:
@jmg-duarte agreed on the naming - `--dry-run` was misleading since it doesn't actually run the service, just validates startup. renamed to `--validate-startup` which is clearer about what it does.

on the negative tests - i added them to catch regressions in error handling (making sure services actually fail loudly on bad config instead of silently proceeding). figured it's worth documenting expected failure behavior, but curious why you think they shouldn't be there - is it the runtime cost, or something else?
